src/PySpectrometer2.py

```diff
@@ -188,18 +193,57 @@ def handle_mouse(event,x,y,flags,param):
 tens = (graticuleData[0])
 fifties = (graticuleData[1])
 
+# load the banner image once
+banner_image = base64.b64decode(background)
+banner_image = np.frombuffer(banner_image, np.uint8)
+banner_image = cv2.imdecode(banner_image, 3)
+
+# the background is fixed to 800x80, so we need to scale it if
+# frameWidth and frameHeight have changed
+banner_image_resized = np.zeros([banner_image.shape[0], frameWidth, 3], dtype=np.uint8)
+w1 = banner_image.shape[1]
+w2 = banner_image_resized.shape[1]
+xoff = round((w2-w1)/2)
+# center the background img in the newly sized background
+banner_image_resized[:, xoff:xoff+w1,:] = banner_image
+banner_image = banner_image_resized
+
+spectrum_vertical = None
+waterfall_vertical = None
+
+def build_graph_base():
+	# blank image for Graph, filled white
+	result = np.full([320, frameWidth, 3], fill_value=255, dtype=np.uint8)
+	textoffset = 12
+
+	# vertial lines every whole 10nm
+	for position in tens:
+		cv2.line(result,(position,15),(position,320),(200,200,200),1)
+
+	# vertical lines every whole 50nm
+	for positiondata in fifties:
+		cv2.line(result,(positiondata[0],15),(positiondata[0],320),(0,0,0),1)
+		cv2.putText(result,str(positiondata[1])+'nm',(positiondata[0]-textoffset,12),font,0.4,(0,0,0),1, cv2.LINE_AA)
+
+	# horizontal lines
+	for i in range (320):
+		if i >= 64 and i%64 == 0: # suppress the first line then draw the rest...
+			cv2.line(result,(0,i),(frameWidth,i),(100,100,100),1)
+			
+	return result
+
+graph_base = build_graph_base()
```

Comment:
Build the background of the display once, this way we can use `np.copy` to just overwrite the display at the start of each loop which is much faster and uses less memory.